src/stores/useSocialMediaStore/useSocialMediaStore.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { act, renderHook } from '@testing-library/react';
+
+import * as useSocialMediaStoreHook from '~stores/useSocialMediaStore/useSocialMediaStore';
+
+vi.mock('~services/api/social-media/social-media', () => ({
+  SocialMediaService: {
+    fetch: vi.fn().mockResolvedValue([
+      { icon: 'discord-icon', id: 'DISCORD_EXAMPLE_ID', name: 'Discord' },
+      { icon: 'twitter-icon', id: 'TWITTER_EXAMPLE_ID', name: 'Twitter' },
+    ]),
+  },
+}));
+
+vi.mock('~services/api/accounts/accounts', () => ({
+  AccountsService: {
+    fetchAll: vi.fn().mockResolvedValue([
+      {
+        avatar: 'https://example.com/image1.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 1,
+        socialMediaId: 'DISCORD_EXAMPLE_ID',
+        token: 'DISCORD_EXAMPLE_TOKEN_1',
+        userName: 'Discord User 1',
+        valid: true,
+      },
+      {
+        avatar: 'https://example.com/image2.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 14,
+        socialMediaId: 'TWITTER_EXAMPLE_ID',
+        token: 'TWITTER_EXAMPLE_TOKEN_14',
+        userName: 'Twitter User 14',
+        valid: true,
+      },
+    ]),
+  },
+}));
+
+/*
+The normal behavior of getAllAccount is "automatically" called when the store is consumed.
+
+In the store file the getAllAccount dont mock when called, then the reccomend way
+to make is, manually, call it to be able to mock on the tests below. 
+
+You can check here:
+https://github.com/pmndrs/zustand/discussions/2563#discussioncomment-9489004
```

Comment:
"Normal" é uma palavra ruim, não significa nada na pratica, no caso vc quer dizer sobre o comportamento esperado.

Ó vamos usar esse texto aqui:
"The expected behavior of the useSocialMediaStore hook is that, when its called, it calls the getAllAccounts method and it happens when the hook is used on the components.

On the tests we need to manually call for getAllAccounts on the renderHook to force its call and mock its return.

The way we're doing the store initialization was based on this thread: https://github.com/pmndrs/zustand/discussions/2563#discussioncomment-9489004"